lib/postgres_v3_experimental.dart

```diff
@@ -1,25 +1,30 @@
 import 'dart:async';
-import 'dart:convert';
 import 'dart:io';
 
 import 'package:collection/collection.dart';
 import 'package:stream_channel/stream_channel.dart';
 
 import 'src/v3/connection.dart';
+import 'src/v3/pool.dart';
 import 'src/v3/protocol.dart';
 import 'src/v3/query_description.dart';
 import 'src/v3/types.dart';
 
 export 'src/v3/types.dart';
 
 abstract class PgPool implements PgSession, PgSessionExecutor {
-  static Future<PgPool> open(
+  static PgPool open(
```

Comment:
The name suggests to me that the connection will be opened/used, and because of that, I think it is worth to keep the `Future` here, e.g. to check if the connection or the credentials are valid.